cdxev/build_public_bom.py

```diff
@@ -72,17 +72,24 @@ def remove_component_tagged_internal(
         "internal"
     """
     # create validator, to check if a component is internal
-    validator_for_being_internal = create_internal_validator(path_to_schema)
     list_of_removed_component_bom_refs = []
     cleared_components = []
-    for component in components:
-        # if it is a internal component, the whole component will be removed,
-        # if not, only the internal information in properties will be removed
-        if validator_for_being_internal.is_valid(component):
-            list_of_removed_component_bom_refs.append(component.get("bom-ref", ""))
-        else:
+
+    if path_to_schema is not None:
+        validator_for_being_internal = create_internal_validator(path_to_schema)
+        for component in components:
+            # if it is a internal component, the whole component will be removed,
+            # if not, only the internal information in properties will be removed
```

Comment:
"only the property within namespace internal" would be the more appropriate description.

Reply:
Changed